src/include/rccl_vars.h

```diff
@@ -25,6 +25,37 @@ THE SOFTWARE.
 
 #include "param.h"
 
-RCCL_PARAM_DECLARE(EnableHipGraph);  // Opt-in environment variable for enabling hipGraph
+RCCL_PARAM_DECLARE(EnableHipGraph);      // Opt-in environment variable for enabling hipGraph
+RCCL_PARAM_DECLARE(ExperimentalXccMode); // Opt-in environment variable for enabling experimental XCC filtering
+
+int32_t __inline__ rcclGetPreferredXcc(int src, int dst)
```

Comment:
The table should be based on src and dst ranks which are unique and not impacted by HIP_VISIBLE_DEVICES and ROCR_VISIBLE_DEVICES